lib/compat.php

```diff
@@ -131,10 +131,22 @@ function gutenberg_ensure_wp_api_request() {
  * @return array           Filtered settings.
  */
 function gutenberg_disable_editor_settings_wpautop( $settings ) {
-	if ( ! isset( $settings['wpautop'] ) ) {
+	if ( is_array( $settings ) ) {
 		$settings['wpautop'] = ! gutenberg_post_has_blocks( get_post()->ID );
 	}
 
 	return $settings;
 }
 add_filter( 'wp_editor_settings', 'gutenberg_disable_editor_settings_wpautop' );
+
+/**
+ * Jetpack Markdown support causes issues when saving a Gutenberg post
+ * by stripping out the <p> tags - this is a bit heavy handed approach
+ * but disabling does fix the issue
+ */
+function gutenberg_remove_jetpack_markdown_support() {
+	if ( defined( 'JETPACK__VERSION' ) ) {
+		remove_post_type_support( 'post', 'wpcom-markdown' );
```

Comment:
`post` is not the only post type we'd expect to be using Gutenberg. Or is it the case that Jetpack is only adding support for this type? Otherwise we might have to hook into where Jetpack is testing the post type and check whether it's handled by Gutenberg. Currently we have a filter for this, but it's not very well generalized:

https://github.com/WordPress/gutenberg/blob/5a949b705ebc0ec3d05b9c98a653c6cd1697a820/lib/register.php#L154

I could imagine a function to call which runs the filter.

We also want to make sure that we're not removing this support if the post is being edited from the classic editor, since this action will be triggered regardless of whether the user is using the Gutenberg editor.